exaudfclient/base/script_options_parser/ctpg/script_option_lines_ctpg.cc

```diff
@@ -0,0 +1,199 @@
+            >= [](skip, auto st, auto ok, skip, auto ov, auto e) { return to_option(ok.get_value(), ov, st.get_sp(), e.get_sp()); },
+        option_value(alpha_numeric)
+            >= [](auto o) { return std::string(o.get_value()); },
+        option_value(option_char)
+            >= [](auto o) { return std::string(o.get_value()); },
+        option_value(whitespaces)
+            >= [](auto o) { return std::string(o.get_value()); },
+        option_value(semicolon_escape)
+            >= [](auto o) { return std::string(";"); },
+        option_value(option_value, option_char)
+            >= [](auto&& ov, auto v) { return std::move(ov.append(v.get_value())); },
+        option_value(option_value, semicolon_escape)
+            >= [](auto&& ov, auto v) { return std::move(ov.append(";")); },
+        option_value(option_value, start_option_token)
+            >= [](auto&& ov, auto v) { return std::move(ov.append("%")); },
+        option_value(option_value, alpha_numeric)
+            >= [](auto&& ov, auto v) { return std::move(ov.append(v.get_value())); },
+        option_value(option_value, whitespaces)
+            >= [](auto&& ov, auto v) { return std::move(ov.append(v.get_value())); },
+        rest(alpha_numeric)
+            >= [](auto r) { return 0;},
+        rest(whitespaces)
+            >= [](auto r) { return 0;},
+        rest(semicolon_escape)
+            >= [](auto r) { return 0;},
+        rest(end_option_token)
+            >= [](auto r) { return 0;},
+        rest(option_char)
+            >= [](auto r) { return 0;},
+        rest(rest, alpha_numeric)
+            >= [](auto r, skip) { return 0;},
+        rest(rest, option_char)
+            >= [](auto r, skip) { return 0;},
+        rest(rest, whitespaces)
+            >= [](auto r, skip) { return 0;},
+        rest(rest, end_option_token)
+            >= [](auto r, skip) { return 0;},
+        rest(rest, start_option_token)
+            >= [](auto r, skip) { return 0;}
+    )
+);
+
+void parse(const std::string& code, options_type& result, std::function<void(const char*)> throwException) {
+    std::stringstream error_buffer;
+    auto res = option_parser.parse(
+        parse_options{}.set_skip_whitespace(false),
+        buffers::string_buffer(code.c_str()),
+        error_buffer);
+    if (res.has_value())
+    {
+        result = res.value();
+    }
+    else
+    {
+        std::stringstream ss;
+        ss << "Error parsing script options: " << error_buffer.str();
+        throwException(ss.str().c_str());
+    }
+}
+
+} //namespace ParserInternals
+
+void parseOptions(const std::string& code, options_map_t & result, std::function<void(const char*)> throwException) {
+    std::string line;
+    size_t current_pos = 0;
+
+    do {
+
+        const size_t new_pos = code.find_first_of("\r\n", current_pos);
+        line = code.substr(current_pos, new_pos);
```

Comment:
Is a string view possible here?

Reply:
Not possible, because CTPG parsers `buffers:string_buffer` expects a const char* or std::string.
However, I changed the code to pass a std:string&& to the parser, so that we can save one string copy.